roles/rhsm-subscription/tasks/main.yaml

```diff
@@ -25,15 +26,13 @@
     state: present
     pool: "{{ rhsm_pool }}"
   when: "'Subscribed' not in subscribed.stdout and rhsm_activation_key is defined and rhsm_activation_key != 'None'"
-  register: register_key_result
+  register: register_key_result and ansible_distribution == "RedHat"
```

Comment:
I'd move the ansible_distribution == "RedHat" either to a conditional on a block that contains all the tasks, or move it into a conditional on the role in the playbook.